src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,100 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    private int size = 0;
 
     @Override
     public void add(T value) {
+        if (size == elementData.length) {
+            elementData = grow();
+        }
 
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        try {
+            if (size == elementData.length) {
+                elementData = grow();
+            }
+            System.arraycopy(elementData, index, elementData, index + 1, size - index);
+            elementData[index] = value;
+            size++;
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
we usually don't catch RuntimExceptions - they indicate developers mistake, let's remove this try catch block